pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -206,6 +206,119 @@ cdef class _Timestamp(datetime):
         object freq       # frequency reference
         list _date_attributes
 
+    # --------------------------------------------------------------------
+    # Constructors
+
+    @classmethod
+    def fromordinal(cls, ordinal, freq=None, tz=None):
+        """
+        Timestamp.fromordinal(ordinal, freq=None, tz=None)
+
+        passed an ordinal, translate and convert to a ts
+        note: by definition there cannot be any tz info on the ordinal itself
+
+        Parameters
+        ----------
+        ordinal : int
+            date corresponding to a proleptic Gregorian ordinal
+        freq : str, DateOffset
+            Offset which Timestamp will have
+        tz : str, pytz.timezone, dateutil.tz.tzfile or None
+            Time zone for time which Timestamp will have.
+        """
+        return cls(datetime.fromordinal(ordinal),
+                   freq=freq, tz=tz)
+
+    @classmethod
+    def now(cls, tz=None):
+        """
+        Timestamp.now(tz=None)
+
+        Returns new Timestamp object representing current time local to
+        tz.
+
+        Parameters
+        ----------
+        tz : str or timezone object, default None
+            Timezone to localize to
+        """
+        if is_string_object(tz):
+            tz = maybe_get_tz(tz)
+        return cls(datetime.now(tz))
+
+    @classmethod
+    def today(cls, tz=None):
+        """
+        Timestamp.today(cls, tz=None)
+
+        Return the current time in the local timezone.  This differs
+        from datetime.today() in that it can be localized to a
+        passed timezone.
+
+        Parameters
+        ----------
+        tz : str or timezone object, default None
+            Timezone to localize to
+        """
+        return cls.now(tz)
+
+    @classmethod
+    def utcnow(cls):
+        """
+        Timestamp.utcnow()
+
+        Return a new Timestamp representing UTC day and time.
+        """
+        return cls.now(UTC)
+
+    @classmethod
+    def utcfromtimestamp(cls, ts):
+        """
+        Timestamp.utcfromtimestamp(ts)
+
+        Construct a naive UTC datetime from a POSIX timestamp.
+        """
+        return cls(datetime.utcfromtimestamp(ts))
+
+    @classmethod
+    def fromtimestamp(cls, ts):
+        """
+        Timestamp.fromtimestamp(ts)
+
+        timestamp[, tz] -> tz's local time from POSIX timestamp.
+        """
+        return cls(datetime.fromtimestamp(ts))
+
+    @classmethod
+    def combine(cls, date, time):
+        """
+        Timestamp.combine(date, time)
```

Comment:
I don't think it makes any sense for these to be on the base class. Unless there is a huge performance gain (which you have shown there is not). This is just plain confusing.